aerogear/mobile-cli
# add dmz url

## JIRA 
https://issues.jboss.org/browse/AEROGEAR-2570
## Changes
- [x] added `mobile set --client=<clientid> --name=<valueName> --value=<value>` command
- [x] added `mobile set client <clientName> --patch=<JSON Patch>`

Notes:
- `oc patch` https://docs.openshift.com/container-platform/3.3/cli_reference/basic_cli_operations.html#patch
 
## Verification notes
Note: Its best to verify this with https://github.com/aerogear/mobile-core/pull/73#pullrequestreview-116551997
- checkout this branch locally
- build
- add mobile crd as described on pinned PR
- `./mobile set value -c=myapp-android -n=dmzUrl -v=www.ddd.ssd --namespace=myproject`
- `./mobile set client myapp-android --patch='{"spec": {"dmzUrl": "www.dfgdfgrldorder.com"}}' --namespace=myproject -o=json`
- `./mobile set client myapp-android --patch='{"spec": {"dmzUrl": null}}' -namespace=myproject -o=json`
- `./mobile set client myapp-android --patch='{"spec": {"dmzUrl": "www.anotherexample.com"}}' --namespace=myproject -o=json`
- `./mobile set value -c=myapp-android -n=dmzUrl  --namespace=myproject -o=json` should set it to ""
- `./mobile set value -c=myapp-android -n=dmzUrl -v=empty.com --namespace=myproject -o=json`
- `./mobile set value -c=myapp-android -n=dmzUrl -v=null --namespace=myproject -o=json`
